### DISCUSSION TITLE
Cloudflare Pages: `functions/_middleware.js` is not called

### DISCUSSION THREAD

**Discussion by mthenw** [remix-run/remix#9309] [Q&A] [answered]
I'm not sure if I should ask about it here or in Cloudflare Discord but it feels a bit more relevant here. 

I have a basic Remix (Vite) app created with 
`npm create cloudflare@latest my-remix-app -- --framework=remix`

according to [Cloudflare docs](https://developers.cloudflare.com/pages/functions/middleware/) I should be able to create `functions/_middleware.js` file that is called on every request, but unfortunately it's not the case. This file is not called either locally or on dev.

**Comment by sergiodxa** [comment, marked as answer]
When using Vite, in development wrangler/workerd is not used as runtime so your CF middlewares will not be called. However when deployed to production because it runs in CF runtime it should work.

The Vite team is working on a way to allows apps/frameworks to change the runtime so instead of Node it can use workerd and run in a production-like env locally.

**Reply by mthenw** [reply]
OK, that makes a lot of sense. Thank you for you answer. Looks like I did something wrong earlier because indeed it works on prod (not locally).

**Reply by v2kovac** [reply]
`npm run build && wrangler pages dev ./build/client` basically works to emulate this, although there's a little wonkiness
